### PR TITLE
Added common name to cert tmpl

### PR DESCRIPTION
**Pull request by jpower432**
# Changes
This adds a common name to the certificates generated for the webhook service to
prevent issues with applications that do CN checking vs SAN checking.

This provides compatibilty for Istio v1.5

There should be no impacts because there was previously no CN on the certificate

Another approach would allow users to provide custom CA certs for this serivce

### CODE REVIEW

**Comment by vagababov** [comment]
/ok-to-test
you need to sign the CLA>

**Comment by vagababov** [comment]
Also we might want a test?

**Comment by jpower432** [comment]
@vagababov I signed the CLA after submitting. How would you recommend testing this?

**Comment by vagababov** [comment]
you need to say
@googlebot I signed it!

**Comment by jpower432** [comment]
@googlebot I signed it!

**Comment by jpower432** [comment]
@googlebot I fixed it.

**Comment by jpower432** [comment]
@googlebot I fixed it.

**Comment by vagababov** [comment]
Let's add a test, and I think this should be fine.

**Comment by vagababov** [comment]
There might be no existing ones right now.
But just testing this create function seems pretty tenable.

**Comment by jpower432** [comment]
@vagababov Sounds good. I was able to find this test file in the project and I looked the results of the unit tests job and this appears to have passed. 
https://github.com/knative/pkg/blob/master/webhook/certificates/resources/certs_test.go

Will this suffice?

Thanks!

**Comment by vagababov** [comment]
We need to verify the new field is set somehow.

File: webhook/certificates/resources/certs_test.go [hunk 1]
```diff
@@ -59,6 +59,13 @@ func TestCreateCerts(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	//Verify common name
+	expectedCommonName := "got-the-hook.knative-webhook.svc"
+
+	if diff := cmp.Diff(caParsedCert.CommonName, expectedCommonName); diff != "" {
+                t.Fatalf("Unexpected Cert Common Name (-want +got) : %v", diff)
+        }
```
Comment (mattmoor) [inline webhook/certificates/resources/certs_test.go:67]: Format Go code:
```suggestion
		t.Fatalf("Unexpected Cert Common Name (-want +got) : %v", diff)
	}
```

File: webhook/certificates/resources/certs_test.go [hunk 2]
```diff
@@ -70,6 +77,10 @@ func TestCreateCerts(t *testing.T) {
 		t.Fatalf("Unexpected CA Cert DNS Name (-want +got) : %v", diff)
 	}
 
+	if diff := cmp.Diff(caParsedCert.DNSNames, expectedDNSNames); diff != "" {
+                t.Fatalf("Unexpected CA Cert DNS Name (-want +got) : %v", diff)
+        }
```
Comment (mattmoor) [inline webhook/certificates/resources/certs_test.go:82]: Format Go code:
```suggestion
		t.Fatalf("Unexpected CA Cert DNS Name (-want +got) : %v", diff)
	}
```

File: webhook/certificates/resources/certs_test.go [hunk 3]
```diff
@@ -59,6 +59,13 @@ func TestCreateCerts(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	//Verify common name
```
Comment (vagababov) [inline webhook/certificates/resources/certs_test.go:62]:
```suggestion
	// Verify common name
```

File: webhook/certificates/resources/certs_test.go [hunk 4]
```diff
@@ -59,6 +59,13 @@ func TestCreateCerts(t *testing.T) {
...
+	expectedCommonName := "got-the-hook.knative-webhook.svc"
```
Comment (vagababov) [inline webhook/certificates/resources/certs_test.go:63]:
```suggestion
	const expectedCommonName = "got-the-hook.knative-webhook.svc"
```

File: webhook/certificates/resources/certs_test.go [hunk 5]
```diff
@@ -59,6 +59,13 @@ func TestCreateCerts(t *testing.T) {
...
+
+	if diff := cmp.Diff(caParsedCert.CommonName, expectedCommonName); diff != "" {
```
Comment (vagababov) [inline webhook/certificates/resources/certs_test.go:65]: Strings can be just compared `a != b`.

File: webhook/certificates/resources/certs_test.go [hunk 6]
```diff
@@ -59,6 +59,13 @@ func TestCreateCerts(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// Verify common name
+	const expectedCommonName = "got-the-hook.knative-webhook.svc"
+
+	if diff := cmp.Diff(caParsedCert.CommonName, expectedCommonName); diff != "" {
+                t.Fatalf("Unexpected Cert Common Name (-want +got) : %v", diff)
+        }
```
Comment (mattmoor) [inline webhook/certificates/resources/certs_test.go:67]: Format Go code:
```suggestion
		t.Fatalf("Unexpected Cert Common Name (-want +got) : %v", diff)
	}
```

File: webhook/certificates/resources/certs_test.go (hunk 2 above)
Comment (mattmoor) [inline webhook/certificates/resources/certs_test.go:82]: Format Go code:
```suggestion
		t.Fatalf("Unexpected CA Cert DNS Name (-want +got) : %v", diff)
	}
```

File: webhook/certificates/resources/certs_test.go [hunk 7]
```diff
@@ -59,6 +59,13 @@ func TestCreateCerts(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// Verify common name
+	const expectedCommonName = "got-the-hook.knative-webhook.svc"
+
+	if caParsedCert.CommonName != expectedCommonName {
+                t.Fatalf("Unexpected Cert Common Name (-want +got) : %v", diff)
+        }
```
Comment (mattmoor) [inline webhook/certificates/resources/certs_test.go:67]: Format Go code:
```suggestion
		t.Fatalf("Unexpected Cert Common Name (-want +got) : %v", diff)
	}
```

File: webhook/certificates/resources/certs_test.go (hunk 2 above)
Comment (mattmoor) [inline webhook/certificates/resources/certs_test.go:82]: Format Go code:
```suggestion
		t.Fatalf("Unexpected CA Cert DNS Name (-want +got) : %v", diff)
	}
```

**Comment by jpower432** [comment]
@mattmoor @vagababov Appreciate all of your feedback. All suggestions incorporated and test has passed. Please let me know if I need to change anything else.

File: webhook/certificates/resources/certs_test.go [hunk 8]
```diff
@@ -59,6 +59,13 @@ func TestCreateCerts(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// Verify common name
+	const expectedCommonName = "got-the-hook.knative-webhook.svc"
+
+	if caParsedCert.Subject.CommonName != expectedCommonName {
+		t.Fatalf("Unexpected Cert Common Name %v, wanted %v", caParsedCert.Subject.CommonName, expectedCommonName)
```
Comment (vagababov) [inline webhook/certificates/resources/certs_test.go:66]:
```suggestion
		t.Fatalf("Unexpected Cert Common Name %q, wanted %q", caParsedCert.Subject.CommonName, expectedCommonName)
```

File: webhook/certificates/resources/certs_test.go [hunk 9]
```diff
@@ -70,6 +77,10 @@ func TestCreateCerts(t *testing.T) {
 		t.Fatalf("Unexpected CA Cert DNS Name (-want +got) : %v", diff)
 	}
 
+	if diff := cmp.Diff(caParsedCert.DNSNames, expectedDNSNames); diff != "" {
+		t.Fatalf("Unexpected CA Cert DNS Name (-want +got) : %v", diff)
```
Comment (vagababov) [inline webhook/certificates/resources/certs_test.go:81]:
```suggestion
		t.Fatalf("Unexpected CA Cert DNS Name (-want +got): %s", diff)
```

File: webhook/certificates/resources/certs.go [hunk 10]
```diff
@@ -46,16 +46,17 @@ func createCertTemplate(name, namespace string, notAfter time.Time) (*x509.Certi
...
 		serviceName,
-		serviceName + ".svc",
+		commonName,
 		serviceName + ".svc.cluster.local",
 	}
 
 	tmpl := x509.Certificate{
 		SerialNumber:          serialNumber,
-		Subject:               pkix.Name{Organization: []string{organization}},
+		Subject:               pkix.Name{Organization: []string{organization}, CommonName: commonName},
```
Comment (mattmoor) [inline webhook/certificates/resources/certs.go:59]: nit:
```suggestion
		Subject: pkix.Name{
			Organization: []string{organization},
			CommonName: commonName,
		},
```

**Comment by mattmoor** [comment]
/approve

Generally this LGTM, but a formatting nit.

File: webhook/certificates/resources/certs.go [hunk 11]
```diff
@@ -46,16 +46,20 @@ func createCertTemplate(name, namespace string, notAfter time.Time) (*x509.Certi
...
 	serviceNames := []string{
 		name,
 		serviceName,
-		serviceName + ".svc",
+		commonName,
 		serviceName + ".svc.cluster.local",
 	}
 
 	tmpl := x509.Certificate{
 		SerialNumber:          serialNumber,
```
Comment (mattmoor) [inline webhook/certificates/resources/certs.go:58]: Format Go code:
```suggestion
		SerialNumber: serialNumber,
```

File: webhook/certificates/resources/certs.go [hunk 12]
```diff
@@ -46,16 +46,20 @@ func createCertTemplate(name, namespace string, notAfter time.Time) (*x509.Certi
...
-		Subject:               pkix.Name{Organization: []string{organization}},
+		Subject: pkix.Name{
+			Organization: []string{organization},
+			CommonName: commonName,
```
Comment (mattmoor) [inline webhook/certificates/resources/certs.go:61]: Format Go code:
```suggestion
			CommonName:   commonName,
```

File: webhook/certificates/resources/certs.go (hunk 11 above)
Comment (mattmoor) [inline webhook/certificates/resources/certs.go:58]: Format Go code:
```suggestion
		SerialNumber: serialNumber,
```

**Comment by jpower432** [comment]
@mattmoor thanks! Formatting changes applied, but that removed the lgtm label.

**Comment by mattmoor** [comment]
Yeah, that's why I didn't bother to `/lgtm` before.  Thanks!

/lgtm

**Pull request merged**